src/tracer.js

```diff
@@ -209,7 +209,7 @@ export default class Tracer {
         let internalTags: any = {};
         if (!parent || !parent.isValid) {
```

Comment:
It becomes a bit messy with handling the debug spans as well. I'd prefer this refactor be a part of a separate commit.